### PR TITLE
Added regex pattern for chap password validation

### PR DESCRIPTION
**Pull request by sneharai4**
Signed-off-by: Sneha Rai <sneha.rai@hpe.com>

### CODE REVIEW

**Comment by sneharai4** [comment]
@bhagyashree-sarawate please review.

**Review by bhagyashree-sarawate — COMMENTED** [review]
LGTM

File: helm/charts/hpe-csi-driver/values.schema.json [hunk 1]
```diff
@@ -120,7 +120,8 @@
                     "title": "CHAP password",
                     "description": "Between 12 and 16 characters",
                     "type": "string",
-                    "default": ""
+                    "default": "",
+                    "pattern": "^$|^[a-zA-Z0-9+_)(*^%$#@!]{12,16}$"
```
Comment (raunakkumar) [inline helm/charts/hpe-csi-driver/values.schema.json:124]: The regex parenthesis doesn't seem to be valid.
 If we are looking for alphanumeric only the following regex would suffice
`^$|^[a-zA-Z0-9+_]{12,16}$`

To keep the above regex valid, it needs to be changed to 
`^$|^[(a-zA-Z0-9+_)(*^%$#@!)]{12,16}$` . 
Could you please verify the change

**Review by raunakkumar — CHANGES_REQUESTED** [review]

File: helm/charts/hpe-csi-driver/values.schema.json (hunk 1 above)
Comment (sneharai4) [reply helm/charts/hpe-csi-driver/values.schema.json:124]: I added the regex using this https://github.com/hpe-storage/common-host-libs/blob/master/linux/iscsi.go#L42 as reference.

Comment (sneharai4) [reply helm/charts/hpe-csi-driver/values.schema.json:124]: Also, the round braces are just characters and not an expression.

Comment (raunakkumar) [reply helm/charts/hpe-csi-driver/values.schema.json:124]: Ok. thanks for the clarification. Looks good then

**Review by raunakkumar — APPROVED** [review]

**Pull request merged**